src/cdk/testing/tests/env-independent/shared-specs.ts

```diff
@@ -0,0 +1,514 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {
+  ComponentHarness,
+  ComponentHarnessConstructor,
+  HarnessLoader,
+  TestElement,
+} from '@angular/cdk/testing';
+import {MainComponentHarness} from '../harnesses/main-component-harness';
+import {SubComponentHarness, SubComponentSpecialHarness} from '../harnesses/sub-component-harness';
+
+/**
+ * Tests that should behave equal in testbed and protractor environment.
+ *
+ * To reduce code duplication and ensure tests will act equal
+ * with TestbedHarnessEnvironment and ProtractorHarnessEnvironment, this set of tests is
+ * executed in unit tests and tests.
+ *
+ * @param getHarnessLoaderFromEnvironment env specific closure to get HarnessLoader
+ * @param getMainComponentHarnessFromEnvironment env specific closure to get MainComponentHarness
+ * @param getActiveElementId env specific closure to get active element
+ *
+ * @docs-private
+ */
+export function sharedCdkTestingSpecs(
+  getHarnessLoaderFromEnvironment: () => HarnessLoader,
+  getMainComponentHarnessFromEnvironment: () => Promise<MainComponentHarness>,
+  // Maybe we should introduce HarnessLoader.getActiveElement(): TestElement
+  // then this 3rd parameter could get removed.
+   getActiveElementId: () => Promise<string | null>,
```

Comment:
https://github.com/e-hein/material2/compare/feature/shared-cdk-testing-specs...e-hein:feature/harness-loader-get-active-element

Reply:
I'm not sure that it really makes sense to have this as part of the API for `HarnessLoader` - I'd rather just leave this in the tests

Reply:
I have use cases for this feature outside cdk/testing. I need it in components with complex focus handling. Maybe you too: e.g. testing cdk/focus-trap. But I think, this is another topic so I did separate it completely from emulating tab.

Reply:
I agree that it might be nice to fit a feature like that in _somewhere_, the `HarnessLoader` just doesn't seem like the right place to me, since it isn't related to what the `HarnessLoader` interface does.